userspace/libsinsp/container.cpp

```diff
@@ -74,15 +76,16 @@ bool sinsp_container_manager::remove_inactive_containers()
 			return true;
 		});
 
-		for(unordered_map<string, sinsp_container_info>::iterator it = m_containers.begin(); it != m_containers.end();)
+		auto containers = m_containers.lock();
+		for(auto it = containers->begin(); it != containers->end();)
 		{
 			if(containers_in_use.find(it->first) == containers_in_use.end())
 			{
 				for(const auto &remove_cb : m_remove_callbacks)
 				{
-					remove_cb(m_containers[it->first]);
+					remove_cb((*containers)[it->first]);
 				}
-				m_containers.erase(it++);
+				containers->erase(it++);
```

Comment:
not you, but just fwiw... I prefer `it = containers->erase(it);` when deleting elements in a loop like this. It looks clearer to me.